src/Adapter/Driver/Pdo/Connection.php

```diff
@@ -237,6 +240,9 @@ public function connect()
                     if (isset($charset) && $pdoDriver != 'pgsql') {
                         $dsn[] = "charset={$charset}";
                     }
+                    if (isset($unix_socket)) {
```

Comment:
> You cannot set the unix socket if a hostname or port is passed
I think, PDO can resolve this issue without problems and zend-db code does not needed additional 'if-else' for this case. But, if you think another, I can add some checks for host and port variables.

> This is really only relevant for mysql as well.
In fact, [PDO_PGSQL DSN](http://php.net/manual/en/ref.pdo-pgsql.connection.php) can connect to unix socket too, but libpq-connect used "host" variable for path to socket.
What do you think I need to do? Add here if-condition for mysql-pdo only? Or add special condition for converting $unix_socket to $host variable for PDO_PGSQL?